spring-boot-project/spring-boot/src/main/java/org/springframework/boot/diagnostics/analyzer/MissingWebServerFactoryBeanFailureAnalyzer.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2012-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.diagnostics.analyzer;
```

Comment:
I'll need to double-check, but this might cause a package tangle. I think it would probably be better if the analyzer went in the same package as the new exception.

Reply:
The same as above.